packages/telegram-stamper/src/index.ts

```diff
@@ -0,0 +1,185 @@
+        this.cloudStorageKeySuffix = "_" + config.cloudStorageKeySuffix;
+      }
+    }
+  }
+
+  // initialize the telegram stamper by getting/setting the private/public api key values from/to telegram cloud storage
+  async init() {
+    // check to see if were in a telegram mini app context
+    this.checkTelegramContext();
+
+    if (this.stamper) {
+      // insert creds into telegram cloud storage
+      await window.Telegram.WebApp.CloudStorage.setItem(
+        this.getCloudStorageKeyName(),
+        this.stringifyAPIKey(),
+        (err: any, stored: boolean) => {
+          if (err != null || !stored) {
+            throw new TelegramStamperError(
+              `Failed inserting api key into Telegram Cloud Storage${
+                err && `: ${err}`
+              }`
+            );
+          }
+        }
+      );
+    } else {
+      // attempt to get creds from telegram cloud storage
+      await window.Telegram.WebApp.CloudStorage.getItem(
+        this.getCloudStorageKeyName(),
+        (err: any, apiKey: string) => {
+          if (err != null || !apiKey) {
+            throw new TelegramStamperError(
+              `Failed getting api key from Telegram Cloud Storage${
+                err && `: ${err}`
+              }`
+            );
+          }
+
+          const { parsedPublicKey, parsedPrivateKey } =
+            this.parseAPIKey(apiKey);
+
+          // ToDo: api key validation, how does apikeystamper validate creds passed to it?'
+
+          if (!parsedPublicKey || !parsedPrivateKey) {
+            throw new TelegramStamperError(
+              "Failed parsing api key from Telegram Cloud Storage"
+            );
+          }
+
+          // instantiate api key stamper
+          this.stamper = new ApiKeyStamper({
+            apiPublicKey: parsedPublicKey,
+            apiPrivateKey: parsedPrivateKey,
+          });
+        }
+      );
+    }
+  }
+
+  async stamp(payload: string) {
+    // check to see if were in a telegram mini app context
+    this.checkTelegramContext();
+
+    // check to see that the stamper was initialized
+    if (!this.stamper) {
+      throw new TelegramStamperError(
+        "Cannot stamp with unintialized telegram stamper, call TelegramStamper.init()"
+      );
+    }
+
+    return await this.stamper.stamp(payload);
+  }
+
+  // clear key from telegram cloud storage
+  async clearKey(key: string) {
+    // check to see if were in a telegram mini app context
+    this.checkTelegramContext();
+
+    await window.Telegram.WebApp.CloudStorage.removeItem(
+      key,
+      (err: any, removed: boolean) => {
+        if (err || !removed) {
+          throw new TelegramStamperError(
+            `Failed removing key: ${key}${err && `: ${err}`}`
+          );
+        }
+      }
+    );
+  }
+
+  checkTelegramContext() {
+    if (window?.Telegram?.WebApp == null) {
+      throw new TelegramStamperError(
+        "Cannot use telegram stamper in non telegram mini-app environment"
```

Comment:
This message should be more specific than this and state that `Telegram.WebApp` is not defined in the current environment. Otherwise users will have to wonder how we determined that the code was run "in a non telegram mini-app environment"

Reply:
updated this comment